### PR TITLE
Make GroupByHash faster

### PR DESCRIPTION
**Pull request by raghavsethi**
Partially addresses #4019.

### CODE REVIEW

**Comment by raghavsethi** [comment]
I'm going to convert both of them to support sizes > INT_MAX later, want to get this in and see perf impact first.

File: presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java [hunk 1]
```diff
@@ -255,14 +258,18 @@ private int addNewGroup(int hashPosition, long value)
...
         return groupId;
     }
 
-    private void rehash(int size)
+    private void rehash()
     {
-        int newSize = arraySize(size + 1, FILL_RATIO);
+        long newSizeLong = hashSize * 2L;
+        if (newSizeLong > Integer.MAX_VALUE) {
+            throw new PrestoException(INSUFFICIENT_RESOURCES, "Size of hash table cannot exceed 2,147,483,647");
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java]: Just say "2 billion entries"

File: presto-main/src/main/java/com/facebook/presto/operator/MultiChannelGroupByHash.java [hunk 2]
```diff
@@ -304,9 +306,13 @@ private void startNewPage()
         }
     }
 
-    private void rehash(int size)
+    private void rehash()
     {
-        int newSize = arraySize(size + 1, FILL_RATIO);
+        long newSizeLong = this.groupIdsByHash.length * 2L;
+        if (newSizeLong > Integer.MAX_VALUE) {
+            throw new PrestoException(INSUFFICIENT_RESOURCES, "Size of hash table cannot exceed 2,147,483,647");
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/operator/MultiChannelGroupByHash.java]: same here

File: presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java [hunk 3]
```diff
@@ -255,14 +258,18 @@ private int addNewGroup(int hashPosition, long value)
...
         return groupId;
     }
 
-    private void rehash(int size)
+    private void rehash()
     {
-        int newSize = arraySize(size + 1, FILL_RATIO);
+        long newSizeLong = hashSize * 2L;
+        if (newSizeLong > Integer.MAX_VALUE) {
+            throw new PrestoException(INSUFFICIENT_RESOURCES, "Size of hash table cannot exceed 2 billion entries");
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java]: `Capacity` instead of `Size`. 1 billion entries instead of 2.

**Comment by haozhun** [comment]
Looks good

**Pull request merged**